lib/Doctrine/DBAL/Types/ObjectType.php

```diff
@@ -58,9 +60,15 @@ public function convertToPHPValue($value, AbstractPlatform $platform)
         }
 
         $value = (is_resource($value)) ? stream_get_contents($value) : $value;
-        $val = unserialize($value);
-        if ($val === false && $value !== 'b:0;') {
-            throw ConversionException::conversionFailed($value, $this->getName());
+
+        set_error_handler(function (int $errno, string $errstr) use ($value) {
+            throw ConversionException::conversionFailedUnserialization($value, $this->getName(), $errstr);
+        });
+
+        try {
+            $val = unserialize($value);
```

Comment:
No need for `$val` variable, you should use `return unserialize(...)` directly here.